Conflicting nullability specifier warnings

There are couple of warnings when bridging the library with Swift 3 (Xcode 8)

```
AppAuth-iOS/Source/OIDAuthState.h:144:1: warning: conflicting nullability specifier on return types, 'nullable' conflicts with existing specifier 'nonnull'
- (nullable instancetype)init NS_UNAVAILABLE;
^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.1.sdk/usr/include/objc/NSObject.h:60:1: note: previous declaration is here
- (instancetype)init
^

AppAuth-iOS/Source/OIDAuthorizationResponse.h:106:1: warning: conflicting nullability specifier on return types, 'nullable' conflicts with existing specifier 'nonnull'
- (nullable instancetype)init NS_UNAVAILABLE;
^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.1.sdk/usr/include/objc/NSObject.h:60:1: note: previous declaration is here
- (instancetype)init
^

AppAuth-iOS/Source/OIDAuthorizationService.h:84:1: warning: conflicting nullability specifier on return types, 'nullable' conflicts with existing specifier 'nonnull'
- (nullable instancetype)init NS_UNAVAILABLE;
^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.1.sdk/usr/include/objc/NSObject.h:60:1: note: previous declaration is here
- (instancetype)init
^

AppAuth-iOS/Source/OIDServiceDiscovery.h:358:1: warning: conflicting nullability specifier on return types, 'nullable' conflicts with existing specifier 'nonnull'
- (nullable instancetype)init NS_UNAVAILABLE;
^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.1.sdk/usr/include/objc/NSObject.h:60:1: note: previous declaration is here
- (instancetype)init
^

AppAuth-iOS/Source/OIDTokenRequest.h:114:1: warning: conflicting nullability specifier on return types, 'nullable' conflicts with existing specifier 'nonnull'
- (nullable instancetype)init NS_UNAVAILABLE;
^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.1.sdk/usr/include/objc/NSObject.h:60:1: note: previous declaration is here
- (instancetype)init
^

AppAuth-iOS/Source/OIDTokenResponse.h:97:1: warning: conflicting nullability specifier on return types, 'nullable' conflicts with existing specifier 'nonnull'
- (nullable instancetype)init NS_UNAVAILABLE;
^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.1.sdk/usr/include/objc/NSObject.h:60:1: note: previous declaration is here
- (instancetype)init
^
```

Some other initializers also have questionable nullability specifiers, e.g.
OIDTokenRequest:
```
- (nullable instancetype)initWithConfiguration:(OIDServiceConfiguration *)configuration …
```

This initializer should be marked as nullable only when there are some circumstances where the initializer can return nil, which is not the case here. This initializer always returns value from NSObject’s init, which never returns nil:
https://developer.apple.com/reference/objectivec/nsobject/1418641-init

With the current annotations, many initializers are marked as “failable” in Swift, e.g.:
let tokenRequest = OIDTokenRequest(…)
initializes token request as optional and requires unnecessary unwrapping

--- a/Source/OIDAuthState.h
+++ b/Source/OIDAuthState.h
@@ -134,7 +134,7 @@ typedef void (^OIDAuthStateAuthorizationCallback)(OIDAuthState *_Nullable authSt
 /*! @internal
     @brief Unavailable. Please use @c initWithAuthorizationResponse:.
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief Creates an auth state from an authorization response.
     @param response The authorization response.

--- a/Source/OIDAuthState.m
+++ b/Source/OIDAuthState.m
@@ -152,7 +152,7 @@ static const NSUInteger kExpiryTimeTolerance = 60;
 
 #pragma mark - Initializers
 
-- (nullable instancetype)init
+- (nonnull instancetype)init
     OID_UNAVAILABLE_USE_INITIALIZER(@selector(initWithAuthorizationResponse:tokenResponse:));
 
 /*! @brief Creates an auth state from an authorization response.

--- a/Source/OIDAuthorizationResponse.h
+++ b/Source/OIDAuthorizationResponse.h
@@ -92,7 +92,7 @@ NS_ASSUME_NONNULL_BEGIN
 /*! @internal
     @brief Unavailable. Please use initWithParameters:.
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief Designated initializer.
     @param request The serviced request.

--- a/Source/OIDAuthorizationResponse.m
+++ b/Source/OIDAuthorizationResponse.m
@@ -109,7 +109,7 @@ static NSString *const kTokenExchangeRequestException =
 
 #pragma mark - Initializers
 
-- (nullable instancetype)init
+- (nonnull instancetype)init
     OID_UNAVAILABLE_USE_INITIALIZER(@selector(initWithRequest:parameters:));
 
 - (nullable instancetype)initWithRequest:(OIDAuthorizationRequest *)request

--- a/Source/OIDAuthorizationService.h
+++ b/Source/OIDAuthorizationService.h
@@ -84,7 +84,7 @@ typedef void (^OIDRegistrationCompletion)(OIDRegistrationResponse *_Nullable reg
 /*! @internal
     @brief Unavailable. This class should not be initialized.
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief Convenience method for creating an authorization service configuration from an OpenID
         Connect compliant issuer URL.

--- a/Source/OIDFieldMapping.h
+++ b/Source/OIDFieldMapping.h
@@ -46,7 +46,7 @@ typedef _Nullable id(^OIDFieldMappingConversionFunction)(NSObject *_Nullable val
 /*! @internal
     @brief Unavailable. Please use initWithName:type:conversion:.
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief The designated initializer.
     @param name The name of the instance variable the field should be mapped to.

--- a/Source/OIDFieldMapping.m
+++ b/Source/OIDFieldMapping.m
@@ -22,7 +22,7 @@
 
 @implementation OIDFieldMapping
 
-- (nullable instancetype)init
+- (nonnull instancetype)init
     OID_UNAVAILABLE_USE_INITIALIZER(@selector(initWithName:type:conversion:));
 
 - (nullable instancetype)initWithName:(NSString *)name

--- a/Source/OIDRegistrationRequest.h
+++ b/Source/OIDRegistrationRequest.h
@@ -77,7 +77,7 @@ NS_ASSUME_NONNULL_BEGIN
 /*! @internal
     @brief Unavailable. Please use initWithConfiguration
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief Designated initializer.
     @param configuration The service's configuration.

--- a/Source/OIDRegistrationResponse.h
+++ b/Source/OIDRegistrationResponse.h
@@ -108,7 +108,7 @@ extern NSString *const OIDRegistrationClientURIParam;
 /*! @internal
     @brief Unavailable. Please use initWithRequest
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief Designated initializer.
     @param request The serviced request.

--- a/Source/OIDRegistrationResponse.m
+++ b/Source/OIDRegistrationResponse.m
@@ -78,7 +78,7 @@ static NSString *const kAdditionalParametersKey = @"additionalParameters";
 
 #pragma mark - Initializers
 
-- (nullable instancetype)init
+- (nonnull instancetype)init
   OID_UNAVAILABLE_USE_INITIALIZER(@selector(initWithRequest:parameters:));
 
 - (nullable instancetype)initWithRequest:(OIDRegistrationRequest *)request

--- a/Source/OIDServiceDiscovery.h
+++ b/Source/OIDServiceDiscovery.h
@@ -316,7 +316,7 @@ NS_ASSUME_NONNULL_BEGIN
     @brief Unavailable. Please use @c initWithDictionary:error:, @c initWithJSON:error, or the
         @c serviceDiscoveryWithURL:callback: factory method.
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief Decodes a OpenID Connect Discovery 1.0 JSON document.
     @param serviceDiscoveryJSON An OpenID Connect Service Discovery document.

--- a/Source/OIDServiceDiscovery.m
+++ b/Source/OIDServiceDiscovery.m
@@ -75,7 +75,7 @@ static NSString *const kOPTosURIKey = @"op_tos_uri";
   NSDictionary *_discoveryDictionary;
 }
 
-- (nullable instancetype)init OID_UNAVAILABLE_USE_INITIALIZER(@selector(initWithDictionary:error:));
+- (nonnull instancetype)init OID_UNAVAILABLE_USE_INITIALIZER(@selector(initWithDictionary:error:));
 
 - (nullable instancetype)initWithJSON:(NSString *)serviceDiscoveryJSON error:(NSError **)error {
   NSData *jsonData = [serviceDiscoveryJSON dataUsingEncoding:NSUTF8StringEncoding];

--- a/Source/OIDTokenRequest.h
+++ b/Source/OIDTokenRequest.h
@@ -99,7 +99,7 @@ NS_ASSUME_NONNULL_BEGIN
     @brief Unavailable. Please use
         initWithConfiguration:grantType:code:redirectURL:clientID:additionalParameters:.
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @param configuration The service's configuration.
     @param grantType the type of token being sent to the token endpoint, i.e. "authorization_code"

--- a/Source/OIDTokenResponse.h
+++ b/Source/OIDTokenResponse.h
@@ -84,7 +84,7 @@ NS_ASSUME_NONNULL_BEGIN
 /*! @internal
     @brief Unavailable. Please use initWithParameters:.
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief Designated initializer.
     @param request The serviced request.

--- a/Source/OIDTokenResponse.m
+++ b/Source/OIDTokenResponse.m
@@ -88,7 +88,7 @@ static NSString *const kAdditionalParametersKey = @"additionalParameters";
 
 #pragma mark - Initializers
 
-- (nullable instancetype)init
+- (nonnull instancetype)init
     OID_UNAVAILABLE_USE_INITIALIZER(@selector(initWithRequest:parameters:));
 
 - (nullable instancetype)initWithRequest:(OIDTokenRequest *)request

--- a/Source/iOS/OIDAuthorizationUICoordinatorIOS.h
+++ b/Source/iOS/OIDAuthorizationUICoordinatorIOS.h
@@ -52,7 +52,7 @@ NS_ASSUME_NONNULL_BEGIN
 /*! @internal
     @brief Unavailable. Please use @c initWithPresentingViewController:
  */
-- (nullable instancetype)init NS_UNAVAILABLE;
+- (nonnull instancetype)init NS_UNAVAILABLE;
 
 /*! @brief The designated initializer.
     @param presentingViewController The view controller from which to present the

